docs/metrics.md

```diff
@@ -0,0 +1,161 @@
+  | thanos_bucket_store_series_gate_queries_max                  | gauge     | Maximum number of concurrent queries.                                                                                                                                               |
+  | thanos_bucket_store_series_get_all_duration_seconds          | histogram | Time it takes until all per-block prepares and loads for a query are finished.                                                                                                      |
+  | thanos_bucket_store_series_merge_duration_seconds            | histogram | Time it takes to merge sub-results from all queried blocks into a single result.                                                                                                    |
+  | thanos_bucket_store_series_refetches                         | counter   | Total number of cases where 65536 bytes was not enough was to fetch series from index, resulting in refetch.                                                                        |
+  | thanos_bucket_store_series_result_series                     | histogram | Number of series observed in the final result of a query.                                                                                                                           |
+  | thanos_bucket_uiblocks_meta_modified                         | gauge     | Number of blocks whose metadata changed.                                                                                                                                            |
+  | thanos_bucket_uiblocks_meta_sync_duration_seconds            | histogram | Duration of the blocks metadata synchronization in seconds.                                                                                                                         |
+  | thanos_bucket_uiblocks_meta_sync_failures                    | counter   | Total blocks metadata synchronization failures.                                                                                                                                     |
+  | thanos_bucket_uiblocks_meta_synced                           | gauge     | Number of block metadata synced.                                                                                                                                                    |
+  | thanos_bucket_uiblocks_meta_syncs                            | counter   | Total blocks metadata synchronization attempts.                                                                                                                                     |
+
+### List of Metrics Exported By Object Storage
+- Thanos leverages object storage solutions such as Minio, Amazon S3, Google Cloud Storage, etc.
```

Comment:
Which component export these metrics?

Reply:
The `thanos_objstore_*` metrics are exported by the Thanos Compactor, I will adjust it to the Compactor PR accordingly.

Reply:
@TomiwaAribisala-git are you sure about this? How did you get to this conclusion?

Reply:
This was an oversight on my part, the `thanos_objstore_*` is not exported by the Compactor; I left you a message on Slack for further reading.